### ISSUE TITLE
New warnings apply to C files when they should only apply to C++ files

### ISSUE THREAD

**Issue by hpvb** [godotengine/godot#28322]
Godot f2d3d3e6791591a572515ea0768bf9c8fb71acd8

@marxin I think the 4134f9e1613e6fea08b5aedd070e4264965df711 needs some fixes to only apply to C++ files as now we get the following warnings:

```
cc1: warning: command line option '-Wctor-dtor-privacy' is valid for C++/ObjC++ but not for C
cc1: warning: command line option '-Wnon-virtual-dtor' is valid for C++/ObjC++ but not for C
cc1: warning: command line option '-Wnoexcept' is valid for C++/ObjC++ but not for C
cc1: warning: command line option '-Wplacement-new=1' is valid for C++ but not for C
```

**Comment by marxin** [comment]
Yes, ideally that warnings should be divided among `CFLAGS` and `CCFLAGS`. Do we want to come up with `CFLAGS`? Adding @akien-mga.

**Comment by akien-mga** [comment]
For SCons, `CCFLAGS` are the flags shared by both C and C++. There is also `CFLAGS` and `CPPFLAGS` for C and C++ respectively.

**Comment by akien-mga** [comment]
Fixed by #28396.
